src/parsing.jl

```diff
@@ -112,20 +112,30 @@ Base.parse(::Type{T}, pqv::PQValue) where {T} = parse(T, string_view(pqv))
 # allow parsing as a Symbol anything which works as a String
 Base.parse(::Type{Symbol}, pqv::PQValue) = Symbol(string_view(pqv))
 
-function Base.start(pqv::PQValue)
-    sv = string_view(pqv)
-    return (sv, start(sv))
-end
+if VERSION >= v"0.7.0-DEV.5125"
+    function Base.iterate(pqv::PQValue)
+        sv = string_view(pqv)
+        iterate(pqv, (sv, firstindex(sv)))
+    end
+    function Base.iterate(pqv::PQValue, (sv, sv_state))
+        iterate(sv, sv_state)
```

Comment:
This generates just the item and the state for `sv`, but it has to put `sv` back in the state that it returns